doc/design/upgrade.md

```diff
@@ -125,15 +135,114 @@ Here is the upgrade workflow
 - Cleanup env after upgrade (remove upgrade snapshot, remove uploaded iso files if upgrade failed)
```

Comment:
Okay, so this is the one version roll-back you mean in the previous comments.
As my understanding, this will be a separate command in vic-machine. Which is to roll-back the upgrade to whatever version it was before upgrade.

And this means most likely user will have left over every time after upgrade, snapshot, image files, etc, to make sure they have opportunity to run one step roll back. And based on this idea, why should we provide --cleanup option for them? Simply lose the opportunity of roll back?

To make the rollback works more cleanly, I will need a cleanup before upgrade, to remove anything left over from last time's upgrade, to avoid confusing in the rollback. And this in turn will not leak more resources just for upgrade.

We could add one cleanup command in the future if this becomes something higher priority.